cmd/skaffold/app/cmd/deploy_test.go

```diff
@@ -17,96 +17,47 @@ limitations under the License.
-		},
-		{
-			description: "one from file, one from CLI",
-			artifacts:   []*latest.Artifact{{ImageName: "image1"}, {ImageName: "image2"}},
-			fromFile:    []build.Artifact{{ImageName: "image1", Tag: "image1:tag"}},
-			fromCLI:     []build.Artifact{{ImageName: "image2", Tag: "image2:tag"}},
-			expected:    []build.Artifact{{ImageName: "image1", Tag: "image1:tag"}, {ImageName: "image2", Tag: "image2:tag"}},
-		},
-		{
-			description: "file takes precedence on CLI",
-			artifacts:   []*latest.Artifact{{ImageName: "image1"}, {ImageName: "image2"}},
-			fromFile:    []build.Artifact{{ImageName: "image1", Tag: "image1:tag"}, {ImageName: "image2", Tag: "image2:tag"}},
-			fromCLI:     []build.Artifact{{ImageName: "image1", Tag: "image1:ignored"}},
-			expected:    []build.Artifact{{ImageName: "image1", Tag: "image1:tag"}, {ImageName: "image2", Tag: "image2:tag"}},
-		},
-		{
-			description: "provide tag for non-artifact",
-			artifacts:   []*latest.Artifact{},
-			fromCLI:     []build.Artifact{{ImageName: "busybox", Tag: "busybox:v1"}},
-			expected:    []build.Artifact{{ImageName: "busybox", Tag: "busybox:v1"}},
-		},
-		{
-			description: "missing tag",
-			artifacts:   []*latest.Artifact{{ImageName: "image1"}, {ImageName: "image2"}},
-			fromFile:    nil,
-			fromCLI:     nil,
+			description: "missing tags",
+			artifacts:   []build.Artifact{{ImageName: "image1"}, {ImageName: "image2"}},
+			expected:    false,
 			shouldErr:   true,
 		},
 		{
-			description: "override tag",
-			artifacts:   []*latest.Artifact{{ImageName: "image1"}, {ImageName: "image2"}},
-			fromFile:    []build.Artifact{{ImageName: "image1", Tag: "image1:tag"}},
-			fromCLI:     []build.Artifact{{ImageName: "image2", Tag: "image2:tag"}},
-			expected:    []build.Artifact{{ImageName: "image1", Tag: "image1:test"}, {ImageName: "image2", Tag: "image2:test"}},
-			customTag:   "test",
+			description: "one missing tag",
+			artifacts:   []build.Artifact{{ImageName: "image1"}, {ImageName: "image2", Tag: "image2:tag"}},
+			expected:    false,
+			shouldErr:   true,
 		},
 		{
-			description: "override missing tag",
-			artifacts:   []*latest.Artifact{{ImageName: "image1"}, {ImageName: "image2"}},
-			fromFile:    nil,
-			fromCLI:     nil,
-			expected:    []build.Artifact{{ImageName: "image1", Tag: "image1:test"}, {ImageName: "image2", Tag: "image2:test"}},
-			customTag:   "test",
+			description: "with tags",
+			artifacts:   []build.Artifact{{ImageName: "image1", Tag: "image1:tag"}, {ImageName: "image2", Tag: "image2:tag"}},
+			expected:    true,
+			shouldErr:   false,
 		},
 	}
 	for _, test := range tests {
 		testutil.Run(t, test.description, func(t *testutil.T) {
-			if test.customTag != "" {
-				t.Override(&opts.CustomTag, test.customTag)
-			}
-
-			deployed, err := getArtifactsToDeploy(ioutil.Discard, test.fromFile, test.fromCLI, test.artifacts)
+			deployed, err := validateArtifactTags(test.artifacts)
```

Comment:
move these tests to `util_test.go` and rename func to `TestValidateArtifacts`

These test cases can be part of `getArtifacts` in util_test.go and you can remove this file.